core/test/element_arithmetic_test.cpp

```diff
@@ -13,16 +14,24 @@ class ElementArithmeticTest : public ::testing::Test {
   const double a = 1.2;
   const double b = 2.3;
   double val = a;
+
+  const int32_t a_int32 = static_cast<int32_t>(1);
+  const int64_t a_int64 = static_cast<int64_t>(1);
+  scipp::core::time_point now = scipp::core::time_point();
 };
 
 TEST_F(ElementArithmeticTest, plus_equals) {
   plus_equals(val, b);
   EXPECT_EQ(val, a + b);
+  EXPECT_NO_THROW(plus_equals(now, a_int32));
+  EXPECT_NO_THROW(plus_equals(now, a_int64));
 }
 
 TEST_F(ElementArithmeticTest, minus_equals) {
   minus_equals(val, b);
   EXPECT_EQ(val, a - b);
+  EXPECT_NO_THROW(minus_equals(now, a_int32));
+  EXPECT_NO_THROW(minus_equals(now, a_int64));
```

Comment:
I think we should not have these tests as part of the tests of `plus_equals` and `minus_equals`. There is no special code for those. Rather, we should add tests for the `time_point` class, then there is no need for testing this here.